bitnami/etcd/templates/scripts-configmap.yaml

```diff
@@ -0,0 +1,184 @@
+{{- $etcdDataDir := include "etcd.dataDir" . }}
+{{- $etcdFullname := include "etcd.fullname" . }}
+{{- $releaseNamespace := .Release.Namespace }}
+{{- $etcdHeadlessServiceName := printf "%s-%s" $etcdFullname "headless" }}
+{{- $dnsBase := .Values.service.dnsBase }}
+{{- $etcdPeerProtocol := include "etcd.peerProtocol" . }}
+{{- $etcdClientProtocol := include "etcd.clientProtocol" . }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "etcd.name" . }}
+    helm.sh/chart: {{ template "etcd.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+  name: {{ template "etcd.fullname" . }}-scripts
+data:
+  setup.sh: |-
+    #!/bin/bash
+
+    set -o errexit
+    set -o pipefail
+    set -o nounset
+
+    # Constants
+    HOSTNAME="$(hostname -s)"
+    AUTH_OPTIONS="{{ $etcdAuthOptions }}"
+    ETCDCTL_ENDPOINTS="{{range $i, $e := until $replicaCount }}{{ $etcdClientProtocol }}://{{ $etcdFullname }}-{{ $e }}.{{ $etcdHeadlessServiceName }}.{{ $releaseNamespace }}.{{ $dnsBase }}:{{ $clientPort }},{{ end }}"
+    # Remove the last comma "," introduced in the string
+    export ETCDCTL_ENDPOINTS="$(sed 's/,/ /g' <<< $ETCDCTL_ENDPOINTS | awk '{$1=$1};1' | sed 's/ /,/g')"
+
+    # Functions
+    ## Store member id for later member replacement
+    store_member_id() {
+        while ! etcdctl $AUTH_OPTIONS member list >/dev/null 2>&1; do sleep 1; done
+        etcdctl $AUTH_OPTIONS member list | grep "$HOSTNAME" | awk '{ print $1}' | awk -F "," '{ print $1}' > "$ETCD_DATA_DIR/member_id"
+        exit 0
+    }
+    ## Configure RBAC
+    configure_rbac() {
+        # When there's more than one replica, we can assume the 1st member
+        # to be created is "{{ $etcdFullname }}-0" since a statefulset is used
+        if [[ -n "$ETCD_ROOT_PASSWORD" ]] && [[ "$HOSTNAME" == "{{ $etcdFullname }}-0" ]]; then
+            echo "==> Configuring RBAC authentication!"
+            etcd > /dev/null 2>&1 &
+            ETCD_PID=$!
+            while ! etcdctl member list >/dev/null 2>&1; do sleep 1; done
+            echo "$ETCD_ROOT_PASSWORD" | etcdctl user add root --interactive=false
+            etcdctl auth enable
+            kill "$ETCD_PID"
+            sleep 5
+        fi
+    }
+    ## Checks whether there was a disaster or not
+    is_disastrous_failure() {
+        local endpoints_array=(${ETCDCTL_ENDPOINTS//,/ })
+        local active_endpoints=0
+        local -r min_endpoints=$((({{ $replicaCount }} + 1)/2))
+
+        for e in "${endpoints_array[@]}"; do
+            if [[ "$e" != "$ETCD_ADVERTISE_CLIENT_URLS" ]] && etcdctl $AUTH_OPTIONS  endpoint health --endpoints="$e" >/dev/null 2>&1; then
+                active_endpoints=$((active_endpoints + 1))
+            fi
+        done
+
+        if [[ -f "/snapshots/.disaster_recovery" ]]; then
+            if [[ $active_endpoints -eq $(({{ $replicaCount }} - 1)) ]]; then
```

Comment:
It may be worth specifying in the docs that scaling using `kubectl scale` will not work (in order to allow that we would need to query `kube-apiserver` for the number of replicas), only `helm upgrade`. It would be interesting exploring `kube-apiserver` access for future iterations of the chart.

Reply:
👍 I'll update the **README.md** warning the users to use `helm upgrade` exclusively